mgmtsystem/hawkular.py

```diff
@@ -766,6 +788,372 @@ def delete_resource(self, feed_id, resource_id):
+                    else:
+                        # recursivelly search in child resources
+                        entities.extend(self._list_child_resource(
+                            entity_value['inventoryStructure']['children']['resource'],
+                            include_data,
+                            resource_type_id,
+                            types_index,
+                            parent_resource_id))
+        return entities
+
+    def _list_child_resource(self, children_j, include_data,
+                             resource_type_id,
+                             types_index,
+                             parent_resource_id):
+        """Returns list of child resources by recursively searching.
+
+         Args:
+            children_j: list of children json objects
+            include_data: whether to include data value of resource
+            resource_type_id: Resource type id
+            types_index: list of typesIndex from inventory
+            parent_resource_id: Id of parent resource
+        """
+        entities = []
+        for child_j in children_j:
+            child_data = child_j['data']
+            if child_data['name'].startswith('{} ['.format(resource_type_id)):
+                # chose those children which name starts with provided resource type id
+                entities.append(ResourceWithData(child_data['id'], child_data['name'],
+                                                 self._get_canonical_path(
+                                                     types_index,
+                                                     child_data['id'],
+                                                     child_data['resourceTypePath'],
+                                                     parent_resource_id),
+                                                 self._get_child_data_value(include_data, child_j)))
+            elif 'resource' in child_j['children']:
+                # otherwise recursively search in children resources
+                entities.extend(self._list_child_resource(
+                    child_j['children']['resource'],
+                    include_data,
+                    resource_type_id,
+                    types_index,
+                    parent_resource_id))
+        return entities
+
+    def get_config_data(self, feed_id, resource_id):
+        """Returns the data/configuration information about resource by provided
+
+        Args:
+            feed_id: Feed id of the resource
+            resource_id: Resource id
+         """
+        if not feed_id or not resource_id:
+            raise KeyError("'feed_id' and 'resource_id' are mandatory field!")
+        result = self._post_raw('strings/raw/query',
+            data={"fromEarliest": "true", "order": "DESC",
+                "tags": "feed:{},type:r,id:{}".format(feed_id, self._get_resource_id(resource_id))})
+        if result.status_code == 200:
```

Comment:
ah, it's not json when code is not 200.
ValueError: No JSON object could be decoded